This patch let's Delly use the system versions of htslib and boost.

Patch by Roel Janssen <roel@gnu.org>
--- a/Makefile	2016-03-16 13:08:49.936706607 +0100
+++ b/Makefile	2016-03-16 13:09:12.351706212 +0100
@@ -9,8 +9,8 @@
 
 # Flags
 CXX=g++
-CXXFLAGS += -isystem ${SEQTK_ROOT} -isystem ${BOOST_ROOT} -pedantic -W -Wall -Wno-unknown-pragmas
-LDFLAGS += -L${SEQTK_ROOT} -L${BOOST_ROOT}/stage/lib -lboost_iostreams -lboost_filesystem -lboost_system -lboost_program_options -lboost_date_time 
+CXXFLAGS += -pedantic -W -Wall -Wno-unknown-pragmas
+LDFLAGS += -lboost_iostreams -lboost_filesystem -lboost_system -lboost_program_options -lboost_date_time 
 
 # Additional flags for release/debug
 ifeq (${PARALLEL}, 1)
@@ -23,7 +23,7 @@
 ifeq (${STATIC}, 1)
 	LDFLAGS += -static -static-libgcc -pthread -lhts -lz
 else
-	LDFLAGS += -lhts -lz -Wl,-rpath,${SEQTK_ROOT},-rpath,${BOOST_ROOT}/stage/lib
+	LDFLAGS += -lhts -lz
 endif
 ifeq (${DEBUG}, 1)
 	CXXFLAGS += -g -O0 -fno-inline -DDEBUG
@@ -41,32 +41,24 @@
 DELLYSOURCES = $(wildcard src/*.h) $(wildcard src/*.cpp)
 
 # Targets
-TARGETS = .htslib .boost src/delly src/extract src/cov src/iover src/stats
+TARGETS = src/delly src/extract src/cov src/iover src/stats
 
 all:   	$(TARGETS)
 
-.htslib: $(HTSLIBSOURCES)
-	cd src/htslib && make && make lib-static && cd ../../ && touch .htslib
-
-.boost: $(BOOSTSOURCES)
-	cd src/modular-boost && ./bootstrap.sh --prefix=${PWD}/src/modular-boost --without-icu --with-libraries=iostreams,filesystem,system,program_options,date_time && ./b2 && ./b2 headers && cd ../../ && touch .boost
-
-src/delly: .htslib .boost $(DELLYSOURCES)
+src/delly: $(DELLYSOURCES)
 	$(CXX) $(CXXFLAGS) $@.cpp -o $@ $(LDFLAGS)
 
-src/extract: .htslib .boost $(DELLYSOURCES)
+src/extract: $(DELLYSOURCES)
 	$(CXX) $(CXXFLAGS) $@.cpp -o $@ $(LDFLAGS)
 
-src/cov: .htslib .boost $(DELLYSOURCES)
+src/cov: $(DELLYSOURCES)
 	$(CXX) $(CXXFLAGS) $@.cpp -o $@ $(LDFLAGS)
 
-src/iover: .htslib .boost $(DELLYSOURCES)
+src/iover: $(DELLYSOURCES)
 	$(CXX) $(CXXFLAGS) $@.cpp -o $@ $(LDFLAGS)
 
-src/stats: .htslib .boost $(DELLYSOURCES)
+src/stats: $(DELLYSOURCES)
 	$(CXX) $(CXXFLAGS) $@.cpp -o $@ $(LDFLAGS)
 
 clean:
-	cd src/htslib && make clean
-	cd src/modular-boost && ./b2 --clean-all
-	rm -f $(TARGETS) $(TARGETS:=.o) .htslib .boost
+	rm -f $(TARGETS) $(TARGETS:=.o)
